test/test_mock.py

```diff
@@ -351,16 +349,48 @@ def test_awaited_from_autospec_mock(self):
         mock = asynctest.mock.create_autospec(Test)
         mock.a_coroutine.assert_not_awaited()
         self.assertFalse(mock.a_coroutine.awaited)
-        self.assertFalse(mock.a_coroutine.awaited.is_set())
         self.assertEqual(0, mock.a_coroutine.await_count)
 
         run_coroutine(mock.a_coroutine())
 
         mock.a_coroutine.assert_awaited()
         self.assertTrue(mock.a_coroutine.awaited)
-        self.assertTrue(mock.a_coroutine.awaited.is_set())
         self.assertEqual(1, mock.a_coroutine.await_count)
 
+    def test_awaited_wait(self):
+        loop = asyncio.new_event_loop()
+        with replace_loop(loop):
```

Comment:
I don't think it is worth changing run_coroutine, which is just a very small helper.
I'd simply use the normal APIs of asyncio and unittest:

    loop = asyncio.new_event_loop()
    self.addCleanup(loop.close)

    mock = asynctest.mock.CoroutineMock()
    t = asyncio.ensure_future(mock.awaited.wait())
    loop.run_until_complete(mock())
    (...)

Reply:
@Martiusweb That's not enough. The new loop is also need to be set and unset (like run_coroutine do) before mock is created, otherwise `asyncio.Condition` will capture the current loop, not the loop that is running it.

Reply:
You're right. We can keep it that way for now, or move these tests in a class extending ``asynctest.TestCase``, since one of its purpose is to handle the loop for each test case.